### PR TITLE
ENH: build with docker-py and Dask

### PR DESCRIPTION
**Pull request by mrakitin**
Uses the following Docker image: https://hub.docker.com/r/mrakitin/debian-with-miniconda/tags/

Requires:
- https://github.com/docker/docker-py
- https://github.com/dask/dask

### CODE REVIEW

File: scripts/nsls2-tag-build.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import os
+import datetime
+import docker
+from pathlib import Path
+
+
+def run_container(*, pkg_name, pythons=('3.5', '3.6', '3.7')):
```
Comment (licode) [inline scripts/nsls2-tag-build.py]: some document here? Maybe docker_image as input?

File: scripts/nsls2-tag-build.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    # Run Docker container
+    docker_client = docker.from_env()
+    output = docker_client.containers.run(docker_image,
+                               name=container_name,
+                               command=f'sh -c "{command}"',
+                               remove=True,
+                               stdout=True,
+                               stderr=True,
+                               volumes=volumes).decode()
```
Comment (licode) [inline scripts/nsls2-tag-build.py]: output to decode?

**Comment by licode** [comment]
good work! some minor suggestion.

File: scripts/nsls2-tag-build.py (hunk 1 above)
Comment (mrakitin) [reply scripts/nsls2-tag-build.py]: Yeah, I'll document the args. Good idea about the docker image.

File: scripts/nsls2-tag-build.py (hunk 2 above)
Comment (mrakitin) [reply scripts/nsls2-tag-build.py]: It's needed since the docker function returns a binary string `b''`.

File: scripts/nsls2-tag-build.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    # Docker run options
+    docker_image = 'mrakitin/debian-with-miniconda:latest'
+    container_name = f'{pkg_name}-{timestamp}'
+    pythons_str = ' '.join(pythons)
+    command = ['./repo/scripts/build.py',
+               f' /repo/recipes-tag/{pkg_name}',
+               f'-u {upload_channel}',
+               f'--python {pythons_str}',
+               '--numpy 1.14',
```
Comment (licode) [inline scripts/nsls2-tag-build.py]: numpy version as input?

File: scripts/nsls2-tag-build.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import os
+import datetime
+import docker
+from pathlib import Path
+
+
+def run_container(*, pkg_name,
+                  docker_image='mrakitin/debian-with-miniconda:latest',
```
Comment (mrakitin) [inline scripts/nsls2-tag-build.py]: Adding this kwarg eventually requires to pass it to the Dask's submitter:
```py
packages = ['bluesky', 'databroker', 'ophyd']
statuses = []
for p in packages:
    st = client.submit(run_container, pkg_name=p, pythons=('3.6',),
                       docker_image='mrakitin/debian-with-miniconda:latest'
                       key=f'{p}-{new_ts()}')
    statuses.append(st)
```

Comment (licode) [reply scripts/nsls2-tag-build.py]: we can ignore it here. or set as default?

Comment (mrakitin) [reply scripts/nsls2-tag-build.py]: No, Dask complains about it if it's not passed, not sure why. Something like:
```py
distributed.worker - WARNING -  Compute Failed
Function:  run_container
args:      ()
kwargs:    {'pkg_name': 'ophyd'}
Exception: TypeError("run_container() missing 1 required keyword-only argument: 'pythons'",)
```

Comment (licode) [reply scripts/nsls2-tag-build.py]: then we can remove docker_image here.

**Comment by licode** [comment]
LGTM. 👍  merge by yourself when you are ready.

**Comment by mrakitin** [comment]
Thanks for the review, @licode!

File: scripts/nsls2-tag-build.py (hunk 4 above)
Comment (mrakitin) [reply scripts/nsls2-tag-build.py]: Yes, it would be reasonable to have it hard-coded to avoid typing it every time.

**Pull request merged**